.github/workflows/docker_build.yml

```diff
@@ -0,0 +1,62 @@
+name: Docker Build
+on:
+  # allow manually trigger
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+      - 'maint-**'
+  pull_request:
+    branches:
+      - main
+      - 'maint-**'
+
+permissions:
+  contents: read
+
+jobs:
+  changes:
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: read
+    outputs:
+      changed: ${{ steps.filter.outputs.ubuntu_build == 'true' }}
+    steps:
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+      
+      - uses: dorny/paths-filter@de90cc6fb38fc0963ad72b210f1f284cd68cea36 # v2
+        id: filter
+        with:
+          filters: .github/path_filters.yml
+
+  build:
+    needs: changes
+    if: ${{ github.repository == 'OpenVisualCloud/Media-Transport-Library' && needs.changes.outputs.changed == 'true' }}
+    runs-on: ubuntu-22.04
+    timeout-minutes: 120
+    permissions:
+      contents: read
+
+    steps:
+      - name: Harden Runner
+        uses: step-security/harden-runner@4d991eb9b905ef189e4c376166672c3f2f230481 # v2.11.0
+        with:
+          egress-policy: audit
+      - name: Checkout repository
+        uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@4fd812986e6c8c2a69e18311145f9371337f27d4 # v3.4.0
+        with:
+          buildkitd-flags: "--debug --allow-insecure-entitlement security.insecure --allow-insecure-entitlement network.host"
+          platforms: "linux/amd64"
+          driver-opts: memory=14Gib,memory-swap=25Gib,env.BUILDKIT_STEP_LOG_MAX_SIZE=50000000,env.BUILDKIT_STEP_LOG_MAX_SPEED=10000000
+
+      - name: Build image
```

Comment:
Have you checked if this work with forks?

Reply:
as it only builds an image, (no login, no pushing) I don't see any reasons why it won't to.

Reply:
`cache-from` only reads from Docker cached layers but the `cache-to:` is totally different thing. First of all, build with this parameter should be lunched and used only when merging to the significant branch - not on every build - as this overrides the current state of art cache file.
Also this part does do the kind of `docker push` and does use both `user` and `token` from user - but this is done under the hood, based on `fithub actor` name and token, check if this does not break anything for frorks as well as make `cache-to` part  to be only executed after merge.

```yaml
61.      cache-from: type=gha,scope=mtl
62.      cache-to: type=gha,mode=max,scope=mtl
```